src/main/java/org/terasology/fluid/system/FluidRegistry.java

```diff
@@ -23,15 +29,33 @@
      * Registers a fluid with a fluid renderer.
      *
      * @param fluidType     The type of fluid
-     * @param fluidRenderer The fluid renderer
+     * @param displayName   The name used for the fluid in the UI
+     * @param fluidTexture  The image to use when rendering the fluid
+     * @param block         The corresponding liquid block, or null if this fluid can't be placed in the world.
      */
-    void registerFluid(String fluidType, FluidRenderer fluidRenderer);
+    void registerFluid(String fluidType, String displayName, BufferedImage fluidTexture, Block block);
+
+    /**
+     * Registers the fluid with no corresponding liquid block and a solid colour.
+     *
+     * @param fluidType   The type of fluid
+     * @param displayName The name used for the fluid in the UI
+     * @param color       The colour to use when rendering the fluid
+     */
+    void registerFluid(String fluidType, String displayName, Color color);
 
     /**
      * Accessor function which returns the list of fluid renderer associated with a given fluid type.
      *
      * @param fluidType The fluid type
      */
-    FluidRenderer getFluidRenderer(String fluidType);
+    BufferedImage getFluidTexture(String fluidType);
+
+    String getDisplayName(String fluidType);
+
+    Block getCorrespondingLiquid(String fluidType);
+
+    String getCorrespondingFluid(Block liquid);
 
+    Prefab getPrefab(String fluidType);
 }
```

Comment:
do we need to add javadocs to these methods?

Reply:
Most of the javadocs existed in the implementation already, so I copied them over to the interface (and added one for getPrefab).